src/sas/mod.rs

```diff
@@ -0,0 +1,190 @@
+    N = b'N',
+    /// A missing value represented using the ASCII letter 'O'.
+    O = b'O',
+    /// A missing value represented using the ASCII letter 'P'.
+    P = b'P',
+    /// A missing value represented using the ASCII letter 'Q'.
+    Q = b'Q',
+    /// A missing value represented using the ASCII letter 'R'.
+    R = b'R',
+    /// A missing value represented using the ASCII letter 'S'.
+    S = b'S',
+    /// A missing value represented using the ASCII letter 'T'.
+    T = b'T',
+    /// A missing value represented using the ASCII letter 'U'.
+    U = b'U',
+    /// A missing value represented using the ASCII letter 'V'.
+    V = b'V',
+    /// A missing value represented using the ASCII letter 'W'.
+    W = b'W',
+    /// A missing value represented using the ASCII letter 'X'.
+    X = b'X',
+    /// A missing value represented using the ASCII letter 'Y'.
+    Y = b'Y',
+    /// A missing value represented using the ASCII letter 'Z'.
+    Z = b'Z',
+}
+
+impl MissingValue {
+    /// Return the ASCII code representation for the missing value.
+    #[inline]
+    pub const fn code(&self) -> u8 {
+        *self as u8
+    }
+}
+
+impl TryFrom<u8> for MissingValue {
+    type Error = &'static str;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            b'.' => Ok(MissingValue::Period),
+            b'_' => Ok(MissingValue::Underscore),
+            b'A' => Ok(MissingValue::A),
+            b'B' => Ok(MissingValue::B),
+            b'C' => Ok(MissingValue::C),
+            b'D' => Ok(MissingValue::D),
+            b'E' => Ok(MissingValue::E),
+            b'F' => Ok(MissingValue::F),
+            b'G' => Ok(MissingValue::G),
+            b'H' => Ok(MissingValue::H),
+            b'I' => Ok(MissingValue::I),
+            b'J' => Ok(MissingValue::J),
+            b'K' => Ok(MissingValue::K),
+            b'L' => Ok(MissingValue::L),
+            b'M' => Ok(MissingValue::M),
+            b'N' => Ok(MissingValue::N),
+            b'O' => Ok(MissingValue::O),
+            b'P' => Ok(MissingValue::P),
+            b'Q' => Ok(MissingValue::Q),
+            b'R' => Ok(MissingValue::R),
+            b'S' => Ok(MissingValue::S),
+            b'T' => Ok(MissingValue::T),
+            b'U' => Ok(MissingValue::U),
+            b'V' => Ok(MissingValue::V),
+            b'W' => Ok(MissingValue::W),
+            b'X' => Ok(MissingValue::X),
+            b'Y' => Ok(MissingValue::Y),
+            b'Z' => Ok(MissingValue::Z),
+            _ => Err("Not a missing value"),
+        }
+    }
+}
+
+/// A 64-bit SAS® floating point number.
+///
+/// SAS® floating point numbers are based on IBM floating point numbers, with
+/// different representations for missing values. When converted to an IEEE-754
+/// floating point number, the missing value representation is encoded as a
+/// valid NaN encoding.
+#[derive(Debug, Copy, Clone)]
+pub struct F64(super::F64);
```

Comment:
I debated creating a named value instead, e.g.:
```rust
pub struct F64 {
    ibm: super::F64,
}
```

The code down below doing `.0.0` gets a bit hard to follow.